code/iaas/ha/src/main/java/io/cattle/platform/ha/monitor/impl/PingInstancesMonitorImpl.java

```diff
@@ -224,6 +226,9 @@ void determineSyncActions(Map<String, KnownInstance> knownInstances, ReportedIns
                 ReportedInstance ri = new ReportedInstance();
                 ri.setExternalId(ki.getExternalId());
                 ri.setUuid(ki.getUuid());
+                Object imageUuid = CollectionUtils.getNestedValue(ki.getData(), DataUtils.FIELDS, FIELD_IMAGE_UUID);
```

Comment:
Craig, do you think we can be more force about certain fields being required for ReportedInstance (in order to avoid possible NPEs in follow up code)? May be present them as the fields in ReportedInstance constructor? Otherwise we might forget to set those fields in the future code where new events are generated.